tests/test_middleware_interface.py

```diff
@@ -228,17 +263,21 @@ def test_ingest_image_fails_missing_file(self):
         self.assertEqual(datasets, [])
 
     def test_run_pipeline(self):
-        # Can't run an actual pipeline because raw/calib/refcat/template data are all zeroed out.
-        with unittest.mock.patch('activator.middleware_interface.SimplePipelineExecutor') as mock_executor:
+        """Test that running the pipeline uses the correct arguments.
+        We can't run an actual pipeline because raw/calib/refcat/template data
+        are all zeroed out.
+        """
+        # Have to setup the data so that we can create the pipeline executor.
+        self.interface.prep_butler(self.next_visit)
+        filename = "fakeRawImage.fits"
+        filepath = os.path.join(self.input_data, filename)
+        data_id, file_data = fake_file_data(filepath,
+                                            self.butler.dimensions,
+                                            self.interface.instrument)
+        with unittest.mock.patch.object(self.interface.rawIngestTask, "extractMetadata") as mock:
+            mock.return_value = file_data
+            self.interface.ingest_image(filename)
+
+        with unittest.mock.patch("activator.middleware_interface.SimplePipelineExecutor.run") as run:
             self.interface.run_pipeline(self.next_visit, {1})
-            mock_executor.from_pipeline.assert_called_once_with(
-                self.interface.pipeline,
-                where=f"detector={self.next_visit.detector} and exposure in (1)",
-                butler=self.interface.butler
-            )
```

Comment:
I think this is a useful test (with maybe some more use of `mock.ANY`), since the restriction to a specific pipeline and data ID is a key part of this method's purpose. It seems a shame to remove it.

Reply:
I'm actually building the pipeline now, though: that's how I discovered some of the bugs (like DM-34202). The only thing being mocked is `run`.

Reply:
Ok, but does that tell you that you built the right executor, or just any executor? The `where` clause seems like it would be hard to test by any means other than direct introspection.